### ISSUE TITLE
hidden(value: Flow) flashes

### ISSUE THREAD

**Issue by rm3dom** [jwstegemann/fritz2#885]
**Describe the bug**

Not really a bug, easy work around. I love fritz2, fritz2 makes me happy, and trying to contribute by reporting this.

Using `hidden` with a `flow` causes flashing, not pretty, but what I do to fix the problem is the following hack:

```
fun Tag<HTMLElement>.hiddenByDefault(value: Flow<Boolean>)
    hidden(true)
    hidden(value)
}
```

There may be other cases where a sane default (early rendered value) could prevent flashing.

**To Reproduce**

Hard to explain, essentially have two div's hidden on some conditions. You need a pretty complex page else the content is rendered so quickly you cant see it.

**Expected behavior**

Hopefully there is a pattern here and when mounting attributes, early defaults could be set until the coroutine loop gets around to it.

**Screenshots**

From this screenshot you can see the contents of the "Settings" tab flashing under the table "Permissions". The tabs at the top maps to flows to hide the content.

![image](https://github.com/user-attachments/assets/6f0c7b3d-7542-4030-a408-0eac1b4f29ef)

With the above hack, the above does not happen.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox / Chrome
 - Version: 130.0 (64-bit)

**Comment by Lysander** [comment]
> Not really a bug, easy work around. I love fritz2, fritz2 makes me happy, and trying to contribute by reporting this.
> 
> Using `hidden` with a `flow` causes flashing, not pretty, but what I do to fix the problem is the following hack:
> 
> ```
> fun Tag<HTMLElement>.hiddenByDefault(value: Flow<Boolean>)
>     hidden(true)
>     hidden(value)
> }
> ```

This is a known and documented behavior, maybe not so easy to find in the docs:
https://www.fritz2.dev/docs/render/#avoid-unsound-values-with-reactive-attributes

I suggest also reading about the special variants we offer especially for styling to avoid such flicker effects:
https://www.fritz2.dev/docs/render/#avoid-flicker-effects-with-reactive-styling

We have discussed about adding some default value for the normal attributes too, but we did not found a good solution, that is not API breaking and also expressive enough. So we decided to live with the *pattern* to set the attribute first statically and in a second call `Flow` based.

So consider this a pattern, not a work around please :-D

As styling is more typical for such effects, we decided to offer there some special `className`-variants.

I hope this helps you :-)

And thx for your contribution - we really appreciate that!

Please close the issue, if you feel this is solved.

**Comment by Lysander** [comment]
PS: The screenshot looks interesting - may I ask, whether the form elements are based upon our headless components? And if so, how do you like those? And if not: why?

**Comment by rm3dom** [comment]
I use most of the headless components, I had to fork the dropdown to make my own combo box, which I see there's a PR for. I think the headless idea is great.

I use the modal and router the most.
I typically build little DSLs around them once I found a pattern as to not repeat myself. 

The project is growing fast I probably now have 22 pages in the Fritz2 SPA and it wraps a kotln react app with another 15 pages. I completely stopped using kotlin react now, I find it horrible compared to Fritz2 now.

The fritz2 Stores and flows binding is incredible, no more double renders, no effects that go into random loops, no crappy hooks. It just works. 

Well done!!!

I missed that part in the docs, closing.
